pkg/rclone/rclone.go

```diff
@@ -312,6 +291,51 @@ func NewRclone(kubeClient *kubernetes.Clientset, port int) Operations {
 	return rclone
 }
 
+// Format from https://rclone.org/rc/#error-returns
+type serverErrorResponse struct {
+	Error  string          `json:"error"`
+	Path   string          `json:"path"`
+	input  json.RawMessage // can contain sensitive info in plain text
+	status int             // same as the http status code
+}
+
+func (s serverErrorResponse) String() string {
+	return fmt.Sprintf(
+		"{%q: %q, %q: %q, %q: %q, %q: %d}",
+		"error",
+		s.Error,
+		"path",
+		s.Path,
+		"input",
+		"<redacted>",
+		"status",
+		s.status,
+	)
+}
+
+func checkResponse(resp *http.Response) error {
+	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+		// everything is ok with the response, there should be no error
+		return nil
+	}
+	body, err := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		// NOTE: do not wrap the error in case it contains sensitive information from the body
+		return fmt.Errorf("could not read the error response body from the rclone server")
+	}
+	var result serverErrorResponse
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		// NOTE: do not wrap the error in case it contains sensitive information from the body
+		return fmt.Errorf("could not unmarshal the error response from the rclone server")
+	}
+	if result.Error == "" {
+		return fmt.Errorf("unmarshalled the response from the server but it had nothing in the error field")
+	}
+	return fmt.Errorf("received error from the rclone server: %s", result.Error)
```

Comment:
question: wouldn't this just serialize the `Error` field instead of using the `String()` method defined on line 302? From what I can see, the `String()` isn't actually used?

Reply:
The `String()` method I added to prevent dumping sensitive info in the logs or sentry - i.e. it is invoked everytime you print the struct (i.e. it is essentially the equivalent of python `__repr__`). I did not want to include the other fields or the whole deserialization of the error in here on purpose. Here I just wanted to show the reason for the error and that reason is in the `Error` field.

Reply:
But if you really prefer to dump the whole struct then I can just call `reslut.String()`.

Reply:
I am dumping the whole struct now via `String()` as you mentioned.